src/pages/Send/index.tsx

```diff
@@ -95,6 +95,15 @@ export default function Send({ location: { search } }: RouteComponentProps) {
   // check whether the user has approved the router on the input token
   const [approval, approveCallback] = useApproveCallbackFromTrade(bestTrade, allowedSlippage)
 
+  // check if user has gone through approval process, used to show two step buttons, reset on token change
+  const [approvalSubmitted, setApprovalSubmitted] = useState<boolean>(false)
+  // mark when a user has submitted an approval, reset onTokenSelection for input field
+  useEffect(() => {
+    if (approval === ApprovalState.PENDING) {
+      setApprovalSubmitted(true)
```

Comment:
is this a state variable so they can swap right after they submit the approve?

Reply:
Yeah we only need this to keep the two step UI around when theyre in this flow - otherwise when the approval changes it will go back to just one swap button (but we want the two step to show up instead here)